osu.Game/Rulesets/Difficulty/Preprocessing/DifficultyHitObject.cs

```diff
@@ -48,6 +61,8 @@ public DifficultyHitObject(HitObject hitObject, HitObject lastObject, double clo
             DeltaTime = (hitObject.StartTime - lastObject.StartTime) / clockRate;
             StartTime = hitObject.StartTime / clockRate;
             EndTime = hitObject.GetEndTime() / clockRate;
+
+            PreviousBacking = new ObjectLink<DifficultyHitObject>(this, null);
```

Comment:
What's the deal with `lastObject` (and `lastLastObject` in `OsuDifficultyHitObject`). 

I would imagine `Previous` to replace them, but I'm not entirely sure about that with the fact that `OsuDifficultyCalculator` creates the first `DifficultyHitObject` starting at the second hitobject, yet sets `LastObject` to the first hitobject.

The xmldoc of `LastObject` should be updated to mention that they can be different from `Previous[0].BaseObject`.

Reply:
I don't understand this comment. What is the discrepancy here? I don't see any.

`lastObject` is the object previous to the current object. Which is what `Previous[0].BaseObject` should be, no? If that's not the case somewhere then it's arguably a bug.

The first difficulty hitobject is created from the first difficulty hitobject because in 99% of cases the difficulty of a note hard-depends on the difficulty of the previous note.

Reply:
It's more about one shouldn't exist if the other is supposed to be equal to it, so `LastObject` should technically be no longer necessary in favour of `Previous[0].BaseObject` (could make it a property: `LastObject => Previous[0].BaseObject`).

But that being said, it's not exactly correct in the case of osu! ruleset, because the first `OsuDifficultyHitObject` (where `Previous = null`) has `BaseObject = objects[1]` and `LastObject = objects[0]`.